src/coreclr/src/tools/Common/System/EnumExtensions.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using ILCompiler.DependencyAnalysis.ReadyToRun;
+using Internal.JitInterface;
+using Internal.ReadyToRunConstants;
+
+namespace System
+{
+    /// <summary>
+    /// Enum.CompareTo takes an Object, so using that requires boxing the enum we are comparing to (which is slow
+    /// and allocates unnecessarily).
+    /// </summary>
+    internal static class EnumExtensions
+    {
+        public static int CompareEnum(this ReadyToRunFixupKind first, ReadyToRunFixupKind second)
+        {
+            return (int)first - (int)second;
+        }
+
+        public static int CompareEnum(this ImportThunk.Kind first, ImportThunk.Kind second)
```

Comment:
If the type is not in Common, code using the type should not be in Common.

+1 to Jan's advice.